src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -419,22 +453,25 @@ namespace AppInstaller::CLI::Workflow
                     context.Reporter.Warn() << Resource::String::MultiplePackagesFound << std::endl;
                 }
 
-                context << ReportSearchResult;
+                context << ReportMultiplePackageFoundResult;
+
                 AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MULTIPLE_APPLICATIONS_FOUND);
             }
 
-            auto latestVersion = searchResult.Matches.at(0).Package->GetLatestAvailableVersion();
-            Logging::Telemetry().LogAppFound(latestVersion->GetProperty(PackageVersionProperty::Name), latestVersion->GetProperty(PackageVersionProperty::Id));
+            auto package = searchResult.Matches.at(0).Package;
+            Logging::Telemetry().LogAppFound(package->GetProperty(PackageProperty::Name), package->GetProperty(PackageProperty::Id));
+
+            context.Add<Execution::Data::Package>(std::move(package));

```

Comment:
>std::move(package) [](start = 50, length = 18)

I don't know the rules around auto clearly enough to say whether this will be a reference or a copy.  I don't think we want to dismantle the SearchResult; that might just lead to bugs.  Can we change the `auto package` to be a clear copy of the Package shared_ptr so that this move is also clearly not taking it from the SearchResult?